File: src/test/java/osmDataRepresentation/osm2ownMaps/PartitionedOsmNodeID2OwnIDMapTest.java
package osmDataRepresentation.osm2ownMaps;

import de.esymetric.jerusalem.ownDataRepresentation.fileSystem.LatLonDir;
import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class PartitionedOsmNodeID2OwnIDMapTest {
    @Test
    public void testPartitionedOsmNodeID2OwnIDMap() {
        LatLonDir.deleteAllLatLonDataFiles("testData");
        de.esymetric.jerusalem.osmDataRepresentation.osm2ownMaps.PartitionedOsmNodeID2OwnIDMap map = new de.esymetric.jerusalem.osmDataRepresentation.osm2ownMaps.PartitionedOsmNodeID2OwnIDMap(
                "testData", false);
        for (int i = 0; i < 25; i++) {
            double lat = Math.random() * 180 - 90;
            double lng = Math.random() * 360 - 180;
            int key = (int) (Math.random() * 788823);
            int value = (int) (Math.random() * 788823548);

            map.put(lat, lng, key, value);
            assertTrue(map.get(key) == value);
        }
        LatLonDir.deleteAllLatLonDataFiles("testData");
    }
}

File: src/test/java/osmDataRepresentation/osm2ownMaps/PartitionedOsmNodeID2OwnIDMapTest.kt
package osmDataRepresentation.osm2ownMaps

import de.esymetric.jerusalem.osmDataRepresentation.osm2ownMaps.PartitionedOsmNodeID2OwnIDMap
import de.esymetric.jerusalem.ownDataRepresentation.fileSystem.LatLonDir.Companion.deleteAllLatLonDataFiles
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class PartitionedOsmNodeID2OwnIDMapTest {
    @Test
    fun testPartitionedOsmNodeID2OwnIDMap() {
        deleteAllLatLonDataFiles("testData")
        val map = PartitionedOsmNodeID2OwnIDMap(
            "testData", false
        )
        for (i in 0..24) {
            val lat = Math.random() * 180 - 90
            val lng = Math.random() * 360 - 180
            val key = (Math.random() * 788823).toInt()
            val value = (Math.random() * 788823548).toInt()
            map.put(lat, lng, key.toLong(), value)
            assertTrue(map[key.toLong()] == value)
        }
        deleteAllLatLonDataFiles("testData")
    }
}